contracts/optimistic-ethereum/OVM/chain/OVM_CanonicalTransactionChain.sol

```diff
@@ -701,22 +755,25 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
     function _getSequencerLeafHash(
         BatchContext memory _context,
         uint256 _nextTransactionPtr,
-        uint256 _txDataLength
+        uint256 _txDataLength,
+        bytes memory _hashMemory
     )
         internal
         pure
         returns (
             bytes32
         )
     {
+        if (BYTES_TILL_TX_DATA + _txDataLength > _hashMemory.length) {
+            _hashMemory = new bytes(BYTES_TILL_TX_DATA + _txDataLength);
+        }
```

Comment:
We should add a test which explicitly tests this codepath. Some batch where there are a bunch of small txs and then one big one

Reply:
Oh yeah, agreed that this is worthwhile.

Additionally, how confident are we that we're correctly passing by reference here?  This would actually be a lot of cost if behind the scenes it was actually allocating new memory to pass downwards.  If it were, we could manually pass in a pointer `uint`.  I can look into this locally and see if it does anything.

Reply:
confirming that pass by ref appears to be working; I did a manual pass by ref with same exact gas properties locally.

Reply:
Gotta make sure that you are always zeroing out the memory before you use it I think. Not sure if this happens automatically. Worth adding a test with a chainElement where there's some 0 bytes and see if they get re-used or are properly zeroed out

Reply:
Sounds good. Will add tests here.